storage/innobase/handler/handler0alter.cc

```diff
@@ -9190,14 +9236,16 @@ innobase_rename_or_enlarge_columns_cache(
 			ulint	col_n = is_virtual ? num_v : i - num_v;
 
 			if ((*fp)->is_equal(cf) == IS_EQUAL_PACK_LENGTH) {
-				if (is_virtual) {
-					dict_table_get_nth_v_col(
-						user_table, col_n)->m_col.len
-					= cf->length;
-				} else {
-					dict_table_get_nth_col(
-						user_table, col_n)->len
-					= cf->length;
+				dict_col_t *col = is_virtual ?
+					&dict_table_get_nth_v_col(
+						user_table, col_n)->m_col
+					: dict_table_get_nth_col(
+						user_table, col_n);
+				col->len = cf->length;
+				if (col->len > 255
+				    && (col->prtype & DATA_MYSQL_TRUE_VARCHAR)
+				    == DATA_MYSQL_TRUE_VARCHAR) {
+					col->prtype |= DATA_LONG_TRUE_VARCHAR;
```

Comment:
I do not think that `innobase_enlarge_column_try()` is updating `SYS_COLUMNS.PRTYPE` at all. The data dictionary cache must be in sync with the data dictionary!

I think that it would be cleaner to remove the old code altogether, and let `innodb_insert_sys_columns()` perform the column size and name changes in `innobase_instant_try()`.

Please add a test that restarts the server (or drops or adds a virtual column, to trigger `dict_table_remove_from_cache()` in `ha_innobase::commit_inplace_alter_table()`).

Reply:
Removed `innobase_rename_columns_try()`.